### PR TITLE
Feature/csh/value changed from

### PR DESCRIPTION
**Pull request by cshcomcom**
## Description
- Now we can select source of VALUE_CHANGED event (`BLOCK`, `USER`)

### CODE REVIEW

File: block-pool/index.js [hunk 1]
```diff
@@ -620,7 +620,8 @@ class BlockPool {
         const stake = _get(validators, `${vote.address}.stake`, 0);
         if (stake > 0) {
           const res = baseDb.executeTransaction(
-              Transaction.toExecutable(vote, chainId), true, true, blockNumber, blockTime);
+              Transaction.toExecutable(vote, chainId), true, true,
+              blockNumber, blockTime, eventSource);
```
Comment (minsulee2) [inline block-pool/index.js:624]: Quadruple spaces?

Comment (minsulee2) [reply block-pool/index.js:624]: Oh, never mind. I got confused.

**Review by minsulee2 — COMMENTED** [review]
Before moving on, I wonder why the source matters?

File: block-pool/index.js [hunk 2]
```diff
@@ -620,7 +620,8 @@ class BlockPool {
         const stake = _get(validators, `${vote.address}.stake`, 0);
         if (stake > 0) {
           const res = baseDb.executeTransaction(
-              Transaction.toExecutable(vote, chainId), true, true, blockNumber, blockTime);
+              Transaction.toExecutable(vote, chainId), true, true,
```
Comment (platfowner) [inline block-pool/index.js:623]: minor) this line seems to be too short after the line wrapping?

File: node/index.js [hunk 3]
```diff
@@ -916,7 +917,7 @@ class BlockchainNode {
    * after removing the invalid transactions from the tx pool.
    */
   executeAndGetValidTransactions(
-      longestNotarizedChain, blockNumber, blockTime, baseDb, isExecutionOnly = false) {
+      longestNotarizedChain, blockNumber, blockTime, baseDb, isExecutionOnly = false, eventSource) {
```
Comment (platfowner) [inline node/index.js:920]: if an argument with a default value is followed by another argument without default values, the default value of the former argument becomes useless.
could you remove the default value?

File: event-handler/index.js [hunk 4]
```diff
@@ -89,6 +92,11 @@ class EventHandler {
         continue;
       }
 
+      const eventSourceConfig = _.get(eventFilter, 'config.event_source', null);
```
Comment (platfowner) [inline event-handler/index.js:95]: minor) eventSourceConfig sound like an object. what about something like expectedEventSource?

File: block-pool/index.js [hunk 5]
```diff
@@ -566,7 +566,7 @@ class BlockPool {
     let tallied = 0;
     for (const vote of lastVotes) {
       if (CommonUtil.isFailedTx(tempDb.executeTransaction(
-          Transaction.toExecutable(vote, chainId), true, true, blockNumber, blockTime))) {
+          Transaction.toExecutable(vote, chainId), true, true, blockNumber, blockTime, null))) {
```
Comment (platfowner) [inline block-pool/index.js:569]: null value skippable?

File: consensus/index.js [hunk 6]
```diff
@@ -771,7 +773,8 @@ class Consensus {
       });
     }
     // Try executing the proposal tx on the proposal block's db state
-    const proposalTxExecRes = tempDb.executeTransaction(executableTx, true, false, number, blockTime);
+    const proposalTxExecRes = tempDb.executeTransaction(executableTx, true, false,
+        number, blockTime, null);
```
Comment (platfowner) [inline consensus/index.js:777]: null value skippable?

File: consensus/index.js [hunk 7]
```diff
@@ -928,7 +931,7 @@ class Consensus {
           `[${LOG_HEADER}] Failed to create a temp state snapshot for vote ${JSON.stringify(executableTx)}`);
       return false;
     }
-    const voteTxRes = tempDb.executeTransaction(executableTx, true, false, prevBlockNumber + 1, prevBlockTime);
+    const voteTxRes = tempDb.executeTransaction(executableTx, true, false, prevBlockNumber + 1, prevBlockTime, null);
```
Comment (platfowner) [inline consensus/index.js:934]: null value skippable?

File: node/index.js (hunk 3 above)
Comment (platfowner) [reply node/index.js:920]: Ah, i think it'd be better to set a default value (null) of eventSource.

File: consensus/index.js [hunk 8]
```diff
@@ -414,10 +415,10 @@ class Consensus {
     }
     const totalAtStake = ConsensusUtil.getTotalAtStake(validators);
     const { offenses, evidence } = this.node.bp.getOffensesAndEvidence(
-        validators, recordedInvalidBlockHashSet, blockNumber, blockTime, tempDb);
+        validators, recordedInvalidBlockHashSet, blockNumber, blockTime, tempDb, null);
     const { transactions, receipts, gasAmountTotal, gasCostTotal } =
         this.node.executeAndGetValidTransactions(
-            longestNotarizedChain, blockNumber, blockTime, tempDb);
+            longestNotarizedChain, blockNumber, blockTime, tempDb, false, null);
```
Comment (platfowner) [inline consensus/index.js:421]: if executeAndGetValidTransactions() has a default value for eventSource, the last two params can be skipped.

**Comment by cshcomcom** [comment]
> Before moving on, I wonder why the source matters?
@minsulee2 
Without this feature, users would not know whether VALUE_CHANGED event was due to the execution of transactions in block or a transaction received from other users. To solve this inconvenience, this feature has been added.

File: block-pool/index.js (hunk 2 above)
Comment (cshcomcom) [reply block-pool/index.js:623]: I agree. I change it to 1 line.

File: node/index.js (hunk 3 above)
Comment (cshcomcom) [reply node/index.js:920]: Looks good. Now we can omit parameters when calling from other places.

File: event-handler/index.js (hunk 4 above)
Comment (cshcomcom) [reply event-handler/index.js]: Looks cool! Done

File: block-pool/index.js (hunk 5 above)
Comment (cshcomcom) [reply block-pool/index.js:569]: Done!

File: consensus/index.js (hunk 6 above)
Comment (cshcomcom) [reply consensus/index.js]: Done!

File: consensus/index.js (hunk 7 above)
Comment (cshcomcom) [reply consensus/index.js:933]: Done!

File: consensus/index.js (hunk 8 above)
Comment (cshcomcom) [reply consensus/index.js]: I make `executeAndGetValidTransactions`'s last 2 parameters to skippable and skip them! Done.

**Comment by cshcomcom** [comment]
All comments are addressed! PTAL @platfowner

**Comment by cshcomcom** [comment]
Thanks for the review! Test completed and now merging!

**Pull request merged**